### PR TITLE
eip-1559

### PR DESCRIPTION
**Pull request by svanas**

### CODE REVIEW

**Comment by juanfranblanco** [comment]
@svanas validated all the test cases for txn type 2, extra parameters for 1559 and everything looks ok

**Review by juanfranblanco — APPROVED** [review]
@svanas validated all the test cases for txn type 2, extra parameters for 1559 and everything looks ok

File: web3.eth.gas.pas [hunk 1]
```diff
@@ -84,12 +111,78 @@ procedure getGasPrice(client: IWeb3; callback: TAsyncQuantity);
...
+
+procedure getMaxPriorityFeePerGas(client: IWeb3; callback: TAsyncQuantity);
+begin
+  var adjustForSpeed := function(tip: BigInteger; speed: TGasPrice): BigInteger
+  begin
+    case speed of
+      Fastest: Result := TWei.Max(tip, 4);
+      Fast   : Result := TWei.Max(tip, 3);
+      Medium : Result := TWei.Max(tip, 2);
+      Low    : Result := 1;
```
Comment (GregTheGreek) [inline web3.eth.gas.pas:133]: I would be hesitant to suggest "speeds".

You're probably best suited to offer a default, and allow overriding it. I think 2.5 is the new default most people use?

File: web3.eth.gas.pas [hunk 2]
```diff
@@ -99,36 +192,74 @@ procedure estimateGas(
...
+          if Assigned(err) then
+            callback(0, err)
+          else
+            eth_estimateGas(client, Format(
+              '{"from": %s, "to": %s, "data": %s, "maxPriorityFeePerGas": %s, "maxFeePerGas": %s}', [
+                web3.json.quoteString(string(from), '"'),
+                web3.json.quoteString(string(&to), '"'),
+                web3.json.quoteString(data, '"'),
+                web3.json.quoteString(toHex(tip, [zeroAs0x0]), '"'),
+                web3.json.quoteString(toHex(maxFee, [zeroAs0x0]), '"')
```
Comment (GregTheGreek) [inline web3.eth.gas.pas:239]: I believe you're missing `accessList`

Comment (svanas) [reply web3.eth.gas.pas:239]: @GregTheGreek does `eth_estimateGas` need `accessList`? I assumed this to be associated with type 2 transaction signing only.

File: web3.eth.gas.pas (hunk 1 above)
Comment (svanas) [reply web3.eth.gas.pas:133]: @GregTheGreek We default to the `eth_maxPriorityFeePerGas` return value but in order to be backwards-compatible with existing dapps offering a low/medium/fast transaction speed, we're post-processing the `eth_maxPriorityFeePerGas` value here. I'm open to your suggestions.

Comment (GregTheGreek) [reply web3.eth.gas.pas:133]: We maintain web3.js, so thats the opinion of the team. I don't think a library should do that

File: web3.eth.gas.pas (hunk 2 above)
Comment (GregTheGreek) [reply web3.eth.gas.pas:239]: my mistake!

**Pull request merged**